crates/jstz_cli/src/sandbox/daemon.rs

```diff
@@ -574,7 +557,7 @@ pub async fn run_sandbox(cfg: &mut Config) -> Result<()> {
     cfg.save()?;
     // 4. Wait for the sandbox or jstz-node to shutdown (either by the user or by an error)
     run_jstz_node(cfg).await?;
-    sandbox.join()?;
+    sandbox.shutdown()?;
```

Comment:
I think this change introduces an issue where if the rollup node / baker or node dies, then the sandbox won't die either -- which was the original intent around `OctezThread::join`

Reply:
Hmm, I'm not sure that even changes before https://github.com/jstz-dev/jstz/pull/525 did that. 
Anyway, I've updated `join` such that it kills all the processes whenever one is killed/dies. To do that, I had make Sandbox have its own local task set that manages both the jstz node and the culling of processes concurrently.